CMakeLists.txt

```diff
@@ -7,26 +7,27 @@ set(CMAKE_CUDA_STANDARD 17)
 set(CMAKE_CUDA_STANDARD_REQUIRED ON)
```

Comment:
Can I change this file in your PR to update this file in order to add `CUDA` as an compile-time option, so that we can get rid of the commented code?

Reply:
Yes sure. 

Also, I found a github action that install a CUDA Toolkit (https://github.com/Jimver/cuda-toolkit) perhaps it can help.